src/pyramid_sanity/__init__.py

```diff
@@ -1,44 +1,19 @@
-"""Main integration into Pyramids automatic configuration hoovering."""
-
-from pyramid.config import PHASE3_CONFIG
-from pyramid.interfaces import ITweens
+from pyramid.tweens import MAIN
 
 from pyramid_sanity._egress import EgressTweenFactory
 from pyramid_sanity._ingress import IngressTweenFactory
 from pyramid_sanity._settings import SanitySettings
 
-__all__ = ["includeme"]
-
-
-def _add_tween(config):
-    sanity_settings = config.registry.settings["pyramid_sanity"]
-    tweens = config.registry.queryUtility(ITweens)
-
-    if sanity_settings.ingress_required:
-        tweens.add_explicit("pyramid_sanity.ingress_tween_factory", IngressTweenFactory)
-
-    if sanity_settings.egress_required:
-        for tween_name, tween_factory in tweens.implicit():
-            tweens.add_explicit(tween_name, tween_factory)
-
-        # Question - `warehouse.sanity` adds this after every individual tween,
-        # but I think it only needs to be in once?
-        tweens.add_explicit(
-            "pyramid_sanity.egress_tween_factory", EgressTweenFactory,
-        )
-
 
 def includeme(config):
-    """Configure `pyramid_sanity` to run when called with `config.include`."""
+    settings = SanitySettings.from_pyramid_settings(config.registry.settings)
+
+    # Put the settings into the registry so that other parts of the
+    # pyramid_sanity code can access them.
+    config.registry.settings["pyramid_sanity"] = settings
 
-    config.registry.settings["pyramid_sanity"] = SanitySettings.from_pyramid_settings(
-        config.registry.settings
-    )
+    if settings.ingress_required:
+        config.add_tween("pyramid_sanity.IngressTweenFactory")
```

Comment:
This gets `under=INGRESS` by default

Reply:
Would it be worth adding to explain intent, or maybe just this as a comment?

Reply:
Also... this is a lot less fuss... but what was the point of all this before? The two options that jump out at me are:

 * It's `warehouse` specific and makes no sense outside
 * It's deeply important in some magical way we'll never understand and the lingering doubt will haunt us to our graves

Reply:
Yeah it's weird and I don't know. I think I prefer doing it in the simple / Pyramid docs way as long as that seems to work / until we learn better, rather than having some complex code we don't understand. I have tested and it seems to work. I came up with a couple more ideas for possible things to test though...

Reply:
Added the comment. Also added something to the `README` about _other tweens_ still being able to crash your app by raising non-ASCII redirects _from the tween_. That's probably not worth fixing. We could use a Pyramid finished callback instead of a tween to ensure that our tween is always called after all other tweens. That might actually fix it. It wouldn't always fix redirects raised by other finished callbacks, but that's pretty obscure. So I might try that but I think it's for a separate PR